scripts/west_commands/runners/nrfjprog.py

```diff
@@ -95,6 +95,17 @@ def do_exec_op(self, op, force=False):
         else:
             raise RuntimeError(f'Invalid operation: {op_type}')
 
+        # Make sure the device is in the expected family
+        try:
+            output = self.check_output(['nrfjprog', '--deviceversion', '--snr', self.dev_id])
```

Comment:
This requires one more execution of the executable, which will affect performance. Can you please benchmark it for both this and the nrfutil runners?

Reply:
This one takes 0,637s on my setup (JLink base compact, nRF91 target)